src/slider/__tests__/slider.test.js

```diff
@@ -0,0 +1,186 @@
+import {mount} from 'enzyme';
+import {Slider, StyledThumb} from '../index';
+
+/* global document */
+/* global window */
+
+const originalDocumentAddListener = document.addEventListener;
+
+describe('Stateless slider', function() {
+  let wrapper;
+  let allProps: {
+      range: Array<number>,
+      value: Array<number>,
+      step: ?number,
+      onChange: () => void,
+    },
+    mockFn;
+
+  beforeEach(function() {
+    //$FlowFixMe
+    document.addEventListener = jest.fn();
+    mockFn = jest.fn();
+    allProps = {
+      onChange: mockFn,
+      range: [0, 100],
+      value: [60],
+      step: 0,
+    };
+  });
+
+  afterEach(function() {
+    jest.restoreAllMocks();
+    document.addEventListener.mockClear();
+    wrapper && wrapper.unmount();
+    //$FlowFixMe
+    document.addEventListener = originalDocumentAddListener;
+  });
+
+  it('should render slider component', function() {
+    wrapper = mount(<Slider {...allProps} />);
+    expect(wrapper).toMatchSnapshot('Component has correctly rendered');
+  });
+
+  describe('events', function() {
+    let event, thumb, step, axisSizeInPixels;
+    function emulateThumbDown(thumbIndex = 0) {
+      wrapper = mount(<Slider {...allProps} />);
+      event = {movementX: 0};
+      thumb = wrapper.find(StyledThumb).at(thumbIndex);
+      thumb.simulate('mousedown', event);
+    }
+    beforeEach(function() {
+      step = 10;
+      axisSizeInPixels = 300;
+      jest
+        .spyOn(window, 'getComputedStyle')
+        .mockImplementation(() => ({width: axisSizeInPixels}));
+      allProps.step = step;
+    });
+
+    it('should start tracking thumb move when user clicks on it', function() {
+      emulateThumbDown();
+      expect(wrapper.instance().state).toMatchObject({
+        isThumbMoving: true,
+        currentThumb: 0,
+      });
+    });
+
+    describe('Thumb Move', function() {
+      const fireMouseMove = event => {
+        const mouseDownCall = document.addEventListener.mock.calls.find(
+          args => args[0] === 'mousemove',
+        )[1];
+        mouseDownCall(event);
+      };
+      it('should move thumb to X pixels forward if mouse is moved', function() {
+        emulateThumbDown();
+        event.movementX = 50;
+        fireMouseMove(event);
+        expect(wrapper.instance().props.onChange).toHaveBeenCalledWith({
+          event,
+          value: [70],
+        });
+      });
+      it('should reset current move progress when thumb moved to X pixels', function() {
+        emulateThumbDown();
+        event.movementX = 50;
+        fireMouseMove(event);
+        expect(wrapper.instance().state).toMatchObject({
+          currentMove: 0,
+        });
+      });
+      it('should note move thumb to X pixels if mouse moved less then step', function() {
+        allProps.step = 50;
+        emulateThumbDown();
+        event.movementX = 20;
+        fireMouseMove(event);
+        expect(wrapper.instance().state).toMatchObject({
+          currentMove: 20,
```

Comment:
Could you explain why currentMove is expected to be set in this case but not in the test case above with almost the same setup?

Reply:
the test name explains it `should not move thumb to X pixels if mouse moved less then step`
so mouse move is 20 and it is less than step=50. I added extra text there